components/camel-cdi/src/main/java/org/apache/camel/cdi/AnnotatedDelegate.java

```diff
@@ -21,11 +21,13 @@
 import java.util.HashSet;
 import java.util.Set;
 
+import javax.enterprise.inject.Vetoed;
```

Comment:
IIRC, we did not used the `@Vetoed` annotation to maintain compatibility with CDI 1.0.

It works because the default profile `weld-3.0` uses CDI 2.0, but other profiles may fail.

That being said, we may want to consider dropping CDI 1.0 support.

Reply:
It does not break CDI 1.0 compatibility since it does not add more beans. Will just enable aries-cdi to drop the beans early and does not affect OWB/Weld/Arc.

Reply:
@astefanutti can you provide your review?

Reply:
@rmannibucau can you confirm `mvn test -Powb-1.0` pass?

Reply:
Indeed it does not but not because of @vetoed, because it was compiled with cdi 2.0 API (bean attributes) so binary is not CDI 1.0 compatible. See org.apache.camel.cdi.CdiCamelBeanRepository#findByTypeWithName which uses javax.enterprise.inject.spi.BeanAttributes#getName for a single example.

side note for that: pom setup breaks multiimpl setup for java > 8 (java 9 profile forces weld).

Reply:
Wow, I see. It's aeecbc47c5accd0425c5cc6c2b5ba8b968d51274 that changed the compilation from using CDI 1.0 to CDI 2.0, and broke binary compatibility as the `Bean` interface changed its hierarchy to use the new `BeanAttributes` interface. So CDI 1.0 compatibility got broken starting Camel 3.0. AFAIK, nobody has ever complained so it may be a good empirical test to considered dropping support safe.